syntax = "proto3";

package app_cpp_server;


service AppCppServer {
    
    rpc get_all (rpcId)  returns (rpcHouse) {};

    rpc notification (rpcId) returns (stream rpcData) {};

    rpc set_state (rpcData) returns (rpcResponse) {};
    
 }
// The request message containing the user's name.

/*message Details {
    string id = 1;
    Data data = 2;
}*/

message rpcData {
    string id = 1;
    string room_name = 2;
    string appliance_type = 3;
    string appliance_name = 4;
    int32 state = 5;
}

// The response message containing the greetings
message rpcState {
    int32 value = 1;
}

message rpcResponse {
    int32 value = 1;
}
message rpcAppliance {
    string app_type = 1;
    string app_name = 2;
    int32 app_state = 3;
}
message rpcRoom {
    string room_name = 1;
    repeated rpcAppliance Appliance = 2;
}

message rpcHouse {
    repeated rpcRoom room = 1;
}

message rpcId {
    string id = 1;
}
